Add support for graceful node shutdown (#12994)
kubernetes/kops: docs/cluster_spec.md

    logFormat: json
```

### Graceful Node Shutdown

Graceful node shutdown allows kubelet to prevent instance shutdown until Pods have been safely terminated or a timeout has been reached.

By default, kOps will try to add a 30 second timeout for 30 seconds where the first 2o seconds is reserved for normal Pods and the last 10 seconds for critical Pods.

This configuration can be changed as follows:

```yaml
spec:
  kubelet:

>     shutdownGracePeriod: 60s
>     shutdownGracePeriodCriticalPods: 20s

Reviewer: To match defaults:
```suggestion
    shutdownGracePeriod: 30s
    shutdownGracePeriodCriticalPods: 10s
```

Author: The point was to show how to change the defaults. so feel like this can/should stay as-is.

> For all CNIs except `amazonaws`, kOpw will try to add a 30 second timeout for 30 seconds where the first 2o seconds is reserved for normal Pods and the last 10 seconds for critical Pods. When using `amazonaws` this feature is disabled, as it leads to [leaking ENIs](https://github.com/aws/amazon-vpc-cni-k8s/issues/1223).

Reviewer: ```suggestion
For all CNIs except `amazonaws`, kOps will try to add a 30 second timeout for 30 seconds where the first 20 seconds is reserved for normal Pods and the last 10 seconds for critical Pods. When using `amazonaws` this feature is disabled, as it leads to [leaking ENIs](https://github.com/aws/amazon-vpc-cni-k8s/issues/1223).
```

```

> Note that Kubelet will fail to install the shutdown inhibtor on systems where logind is configured with an `InhibitDelayMaxSeconds` lower than `shutdownGracePeriod`. On Ubuntu, this setting is 30 seconds.

Reviewer: ```suggestion
Note that Kubelet will fail to install the shutdown inhibtor on systems where `systemd-logind` is configured with an `InhibitDelayMaxSeconds` lower than `shutdownGracePeriod`. On Ubuntu, this setting is 30 seconds.
```
